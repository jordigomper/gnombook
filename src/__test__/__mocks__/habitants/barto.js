export default {
  name: "El Barto",
  age: 14,
  friends: ["Milhouse Van Houten"],
  hair_color: "Unknown",
  height: 107.75835,
  id: 1,
  professions: [],
  thumbnail: null,
  weight: 50.052,
};
