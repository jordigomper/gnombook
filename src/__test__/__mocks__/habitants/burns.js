export default {
  name: "Mr. Burns",
  age: null,
  friends: ["Milhouse Van Houten"],
  hair_color: "gray",
  height: 95.75835,
  id: 4,
  professions: ["Chief"],
  thumbnail:
    "http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg",
  weight: 50.052,
};
