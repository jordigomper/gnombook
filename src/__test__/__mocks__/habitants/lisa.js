export default {
  name: "Lisa Simpson",
  age: 11,
  friends: [],
  hair_color: "Yellow",
  height: 105.95435,
  id: 3,
  professions: ["Student", "Musician"],
  thumbnail:
    "http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg",
  weight: 34.4852,
};
