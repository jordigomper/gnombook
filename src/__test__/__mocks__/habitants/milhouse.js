export default {
  name: "Milhouse Van Houten",
  age: 13,
  friends: ["El Barto"],
  hair_color: "Blue",
  height: 103,
  id: 2,
  professions: ["Student"],
  thumbnail:
    "http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg",
  weight: 39.065952,
};
