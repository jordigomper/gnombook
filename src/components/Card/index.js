import React from "react";
import PropTypes from "prop-types";
import { navigate } from "@reach/router";

import { ImageCache } from "../";
import { Container } from "./elements";

const phantom_gnome = require("../../assets/img/top_secret.jpg");

export const Card = ({ id, thumbnail, name }) => {
  return (
    <Container onClick={() => navigate(`/profile/${id}`)}>
      <ImageCache src={thumbnail} alt={name} title={name} />
      <p>{name}</p>
    </Container>
  );
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  thumbnail: PropTypes.string.isRequired,
};

Card.defaultProps = {
  name: "",
  thumbnail: phantom_gnome,
};
