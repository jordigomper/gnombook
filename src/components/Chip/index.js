import React, { useState } from "react";
import PropTypes from "prop-types";

import { Container } from "./elements";

export const Chip = ({ children, onClick }) => {
  const [isActive, setIsActive] = useState(false);
  const handleClick = () => {
    onClick(children);
    setIsActive(!isActive);
  };
  return (
    <Container onClick={handleClick} isActive={isActive}>
      {children}
    </Container>
  );
};

Chip.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

Chip.propTypes = {
  children: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
