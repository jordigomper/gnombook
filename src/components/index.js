export { Card } from "./Card";
export { FilterPanel } from "./FilterPanel";
export { ImageCache } from "./ImageCache";
export { Searchbar } from "./Searchbar";
export { Chip } from "./Chip";
