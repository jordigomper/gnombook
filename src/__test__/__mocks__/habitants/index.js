import barto from "./barto.js";
import milhouse from "./milhouse.js";
import lisa from "./lisa.js";

export { barto, milhouse, lisa };
