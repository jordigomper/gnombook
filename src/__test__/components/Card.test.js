import React from "react";
import { shallow } from "enzyme";
import { Card } from "../../components";
import { lisa } from "../__mocks__/habitants";

jest.mock("../../assets/img/top_secret.jpg", () => {});

describe("Card component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Card {...lisa} />);
    expect(wrapper).toMatchSnapshot();
  });
});
