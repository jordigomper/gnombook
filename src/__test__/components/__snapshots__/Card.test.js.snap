// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `ShallowWrapper {}`;

exports[`Card component starts with a count of 0 1`] = `ShallowWrapper {}`;
